src/services/homework/hooks/submissions.js

```diff
@@ -6,17 +6,45 @@ const {	iff, isProvider } = require('feathers-hooks-common');
 
 const globalHooks = require('../../../hooks');
 const { equal: equalIds } = require('../../../helper/compare').ObjectId;
+const { submissionModel } = require('../model');
 
 const filterRequestedSubmissions = async (context) => {
 	const currentUserId = context.params.account.userId;
 
 	// user is submitter
 	let permissionQuery = { studentId: currentUserId };
 
-	// user is team member
-	permissionQuery = { $or: [permissionQuery, { teamMembers: { $in: currentUserId } }] };
+	// user is team member and the homework is not private
+	const homeworkService = context.app.service('/homework');
+	const subquery = {
+		$and: [
+			{ teamMembers: { $in: currentUserId } },
+			{ studentId: { $ne: currentUserId } },
```

Comment:
Do we have to exclude the sumitter (`studentId`)? I think the permissions for team member are a subset of the permissions of the submitter? But probably doesn't make much different.

Reply:
Currently the submitter is always automatically added as team member. The check is necessary, otherwise we find all submissions of the current user - which is an overkill for the next query.

Reply:
Currently the submitter is always automatically added as team member. The check is necessary, otherwise we find all submissions of the current user - which is an overkill for the next query.